urlutil.py

```diff
@@ -26,3 +31,18 @@ def get_local_file_from_url(url, tempdir):
             output.write(stream.read())
 
     return pathname
+
+def find_firmware_by_hash(hash, board, language):
+    resource = boto3.resource('s3')
+    resource.meta.client.meta.events.register('choose-signer.s3.*', disable_signing)
```

Comment:
Tell me more about this unfortunate bit of code? Is there some reason we can't download the URL directly from the bucket?

Reply:
This was the way I had to do it in order to fetch data from s3 without authenticating the client. I was kind of weary about adding the `boto3` dependency anyway. Perhaps it's possible to skip that and try to pull data with urllib or something else higher level than this s3 wrapper.